libs/barista-components/filter-field/src/filter-field-multi-select/filter-field-multi-select.ts

```diff
@@ -0,0 +1,344 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ActiveDescendantKeyManager, Highlightable } from '@angular/cdk/a11y';
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
+import { TemplatePortal } from '@angular/cdk/portal';
+// tslint:disable: template-cyclomatic-complexity
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Inject,
+  InjectionToken,
+  Input,
+  NgZone,
+  Output,
+  QueryList,
+  TemplateRef,
+  ViewChild,
+  ViewChildren,
+  ViewContainerRef,
+  ViewEncapsulation,
+} from '@angular/core';
+import { DtCheckboxChange } from '@dynatrace/barista-components/checkbox';
+import { DtOption } from '@dynatrace/barista-components/core';
+import { xor } from 'lodash-es';
+import { merge, Subject } from 'rxjs';
+import { startWith, switchMap, takeUntil } from 'rxjs/operators';
+import { DtNodeDef, isDtGroupDef } from '../types';
+
+let _uniqueIdCounter = 0;
+
+export class DtFilterFieldMultiSelectSubmittedEvent<T> {
+  constructor(
+    /** Reference to the filter field multiSelect panel that emitted the event. */
+    public source: DtFilterFieldMultiSelect<any>,
+    /** Selected option(s) */
+    public multiSelect: T[],
+  ) {}
+}
+
+/** Default `dt-multi-select` options that can be overridden. */
+export interface DtMultiSelectDefaultOptions {
+  /** Whether the first option should be highlighted when an multi-select panel is opened. */
+  autoActiveFirstOption?: boolean;
+}
+
+/** Injection token to be used to override the default options for `dt-multi-select`. */
+export const DT_MULTI_SELECT_DEFAULT_OPTIONS = new InjectionToken<
+  DtMultiSelectDefaultOptions
+>('dt-multi-select-default-options', {
+  providedIn: 'root',
+  factory: DT_MULTI_SELECT_DEFAULT_OPTIONS_FACTORY,
+});
+
+/** @docs-private */
```

Comment:
Has no effect in our repo. please use @internal and prefix it with an `_`
```suggestion
@internal
```